Code/Framework/AzCore/AzCore/RTTI/BehaviorContext.h

```diff
@@ -206,25 +216,53 @@ namespace AZ
     public:
         AZ_CLASS_ALLOCATOR(BehaviorDefaultValue, AZ::SystemAllocator, 0);
 
+        const BehaviorValueParameter& GetValue() const
+        {
+            return m_value;
+        }
+
+        BehaviorValueParameter m_value;
+    };
+
+    template<typename Value>
+    class BehaviorDefaultValueTyped
+        : public BehaviorDefaultValue
+    {
+    public:
         /**
         * Create a default value for a specific method parameter. The Default values is stored by value
         * in a temp storage, so currently there is limit the \ref BehaviorValueParameter temp storage, we
         * can easily change that if it became a problem.
         */
-        template<typename Value>
-        BehaviorDefaultValue(Value&& value)
+        template<typename U, typename = AZStd::enable_if_t<AZStd::is_same_v<Value, AZStd::decay_t<U>>>>
+        BehaviorDefaultValueTyped(U&& value)
```

Comment:
Why have an enable_if_t check that is restrictive?
The only check that should be here is to make sure that U isn't derived from type `BehaviorDefaultValue`.

Besides that any U should be supported as long as it is convertible to `Value`
```suggestion
        template<typename U, typename = AZStd::enable_if_t<AZStd::conjunction_v<
        !AZStd::derived_from<AZStd::decay_t<U>, BehaviorDefaultValue>,
        AZStd::convertible_to<U, Value>>>
        BehaviorDefaultValueTyped(U&& value)
```

Reply:
I tried that, but changing `BehaviorValueParameter` to correctly destroy the type it stores depending on if `StoreInTempData()` was used proved to be really tricky. Given the current use of this type, `BehaviorValueParameter` needs to be copyable and movable, which is not possible because it uses a stack-based allocator to allocate space to store the instance.

Reply:
This is not changing the BehaviorValueParameter, this is changing the BehaviorDefaultValue class which encapsulates a `BehaviorValueParameter`
Having a derived `BehaviorDefaultValueTyped` doesn't change stack based allocator issue of the BehaviorValueParameter